core/commands/UserTokenCreateCommand.php

```diff
@@ -0,0 +1,130 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'authentication_api.php' );
+require_api( 'api_token_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_account_api.php' );
+
+/**
+ * A command that creates a user API token.
+ */
+class UserTokenCreateCommand extends Command {
+	/**
+	 * @var integer The user id.
+	 */
+	private $user_id;
+
+	/**
+	 * @var string The token name.
+	 */
+	private $name;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		// acting user id
+		$t_current_user_id = auth_get_current_user_id();
+
+		// user is to create token for
+		$this->user_id = $this->query( 'user_id' );
+
+		// any user can create a token for themselves, but only an admin can create for other users.
+		if( $t_current_user_id != $this->user_id ) {
```

Comment:
Maybe option `$g_impersonate_user_threshold` has also be considered.
Being able to create a token for another user means being able to act as another user in two ways
- creat a token in the name of the user
- use the token in long term to act as this user

Reply:
Good suggestion. I switched over to use impersonation threshold instead. It makes more sense for the scenario. Thanks.